appinventor/appengine/src/com/google/appinventor/client/editor/youngandroid/YaProjectEditor.java

```diff
@@ -122,6 +123,20 @@ private class EditorSet {
   private boolean screen1BlocksLoaded = false;
   private boolean screen1Added = false;
 
+   // variable which open the ProjectPropertyDialog(per project)
+  private ProjectPropertiesDialogBox propertyDialogBox = null;
+
+  /**
+   * Opens the project property dialog
+   */
+  public void openProjectPropertyDialog() {
+    if (propertyDialogBox == null) {
+      propertyDialogBox = new ProjectPropertiesDialogBox();
+    }
+    String curScreen = Ode.getInstance().getDesignToolbar().getCurrentProject().currentScreen;
+    propertyDialogBox.showDialog(curScreen);
+  }
```

Comment:
Not sure if creating a new dialog each time it's opened is the right way to go. A better approach would be to have a singleton whose contents you can refresh if the project was changed.

Reply:
One thing I found out that, creating instance of dialog on the YaProjectEditor's constructor will generate error cause the MockForm won't  be created and we will access MockForm's Properties through dialog constructor.